source/slang/hlsl.meta.slang

```diff
@@ -812,99 +816,132 @@ float2 GetRenderTargetSamplePosition(int Index);
 __target_intrinsic(glsl, "groupMemoryBarrier")
 void GroupMemoryBarrier();
 
+
 __target_intrinsic(glsl, "groupMemoryBarrier(), barrier()")
+// TODO(JS): I don't think __syncwarp can be used because a 'group' could be multiple warps
+__target_intrinsic(cuda, "__syncthreads()")
 void GroupMemoryBarrierWithGroupSync();
 
 // Atomics
 
 __target_intrinsic(glsl, "$atomicAdd($A, $1)")
+__target_intrinsic(cuda, "atomicAdd(&$0, $1)")
```

Comment:
Having to put `&` in front of `$0` here is a byproduct of the known issue in our C++/CUDA codegen where we emit the `someVar` instead of `&someVar` and so forth for expressions of pointer type.

The fix here is the expedient one, but we will eventually need/want to revisit the original issue in the emit logic.